src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java

```diff
@@ -637,15 +637,21 @@ public static boolean uploadRepo(String repoPath, String repoName, String[] file
             }
             fileUrls = new ObjectMapper().readValue(urls.toJSONString(), new TypeReference<Map<String, Object>>(){});
 
-            codeSyncProgressIndicator.setMileStone(InitRepoMilestones.UPLOAD_FILES);
-            // Upload file to S3.
-            uploadToS3(repoPath, branchName, accessToken, email, repoId, fileUrls);
+            S3FileUploader s3FileUploader = new S3FileUploader(configRepo.repoPath, branchName, fileUrls);
+            s3FileUploader.saveURLs();
+            s3FileUploader.triggerAsyncTask(project);
```

Comment:
How about we only save the YML file and let `processFiles` take care of this upload since `triggerS3Uploads` creates new instance of `s3FilesUploader` so same file  might be in-process by 2 processes (different `filesBeingProcessed` values) 
Kindly correct me if I am missing anything here, thanks